doc/contribute/development_workflow.rst

```diff
@@ -0,0 +1,630 @@
+
+5. Compare the changes with the previous version using ``git diff``.
+   This brings up a simple text browser interface that highlights the difference
+   between your files and the previous version like this:
+
+   .. code::
+
+      diff --git a/development_workflow.md b/development_workflow.md
+      index f05f0cd..e581f00 100644
+      --- a/development_workflow.md
+     +++ b/development_workflow.md
+      @@ -8,17 +8,22 @@ layout: index
+
+6. Inform Git of what modified or new files you want to save (stage) using ``git add modified_file``.
+   This puts the files into a ``staging area``, which is a
+   list of files that will be added to your next commit. Only add files that have
+   related, complete changes. Leave files with unfinished changes for later
+   commits.
+
+7. To commit the staged files into the local copy of your repo, do
+   ``git commit``. Write a clear Git commit message that describes the changes
+   that you have made. Please read `this article <http://chris.beams.io/posts/git-commit/>`_
+   on writing commit messages. If a commit fixes an open issue on the `GitHub issue
+   tracker <https://github.com/nest/nest-simulator/issues>`_, include
+   ``Fixes #issue_number`` in the commit message. GitHub finds such keywords and
+   closes the issue automatically when the pull request is merged. For a list of
+   all keywords you can use, refer to `this GitHub help
+   page <https://help.github.com/articles/closing-issues-via-commit-messages/>`_.
+   After saving your message and closing the editor, your commit will be saved.
+
+8. Push the changes to your forked repo on GitHub:
+
+   .. code::
+
+      git push origin my-new-feature
+
+Assuming you have followed the instructions in these pages, git will create
+a default link to your GitHub repo called ``origin``. In git >= 1.7 you can
+ensure that the link to origin is permanently set by using the ``--set-upstream``
+option:
+
+.. code::
+
+   git push --set-upstream origin my-new-feature
+
+From now on, Git will know that ``my-new-feature`` is related to the
+``my-new-feature`` branch in your own GitHub repo. Subsequent push calls
+are then simplified to the following:
+
+.. code::
+
+   git push
+
+It often happens that while you were working on your edits, new commits have
+been added to ``upstream`` that affect your work. In this case, you will need to
+reposition your commits on the new master. Please follow the instructions on the
+:ref:`rebase` section to see how this is handled.
+
+Next, we see how to create a pull request.
+
+.. _pull_request:
+
+Creating a pull request
+~~~~~~~~~~~~~~~~~~~~~~~
+
+When you feel your work is finished, you can create a pull request (PR). GitHub
+has a nice help page that outlines the process for
+`submitting pull requests <https://help.github.com/articles/using-pull-requests/#initiating-the-pull-request>`_.
+Your pull request will usually be reviewed by other NEST developers using the code review guidelines.
+
+Advanced cases
```

Comment:
I would like to vote for removing the advanced cases altogether. They make the document only longer without a benefit to most people and all core developers should be able to solve these kinds of problems on their own.

Reply:
Yes, I agree and have therefore removed the advanced cases. If anything, the information is also available in the Git documentation.